addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -212,7 +214,116 @@ def _search_virtual_remaining_leaves(self, operator, value):
 
         return [('id', 'in', valid_leave_types.ids)]
 
+    def get_employees_days_per_allocation(self, employee_ids, date=None):
+        leaves = self.env['hr.leave'].search([
+            ('employee_id', 'in', employee_ids),
+            ('state', 'in', ['confirm', 'validate1', 'validate']),
+            ('holiday_status_id', 'in', self.ids)
+        ])
+
+        allocations = self.env['hr.leave.allocation'].search([
+            ('employee_id', 'in', employee_ids),
+            ('state', 'in', ['confirm', 'validate1', 'validate']),
+            ('holiday_status_id', 'in', self.ids),
+        ])
+
+        if not date:
+            date = fields.Date.to_date(self.env.context.get('default_date_from')) or fields.Date.context_today(self)
+
+        allocation_employees = defaultdict(lambda: defaultdict(list))
+
+        ### Creation of the allocation intervals ###
+        for holiday_status_id in allocations.holiday_status_id:
+            for employee_id in employee_ids:
+                allocation_intervals = Intervals([(
+                    fields.datetime.combine(allocation.date_from, time.min),
+                    fields.datetime.combine(allocation.date_to or datetime.date.max, time.max),
```

Comment:
Probably use the max date_to of leaves instead of datetime.max

and don't do all the computation when there are no leaves taken